BBallStatsTrack/src/com/example/bballstatstrack/models/Game.java

```diff
@@ -0,0 +1,406 @@
+        if( mEventGameClock == -1 )
+        {
+            timeDifference = 0;
+        }
+        else
+        {
+            timeDifference = mEventGameClock - mCurrentGameClock;
+        }
+        mCurrentShotClock = MAX_SHOT_CLOCK - timeDifference;
+    }
+
+    public void setTeamWithPossession( Team team )
+    {
+        if( team.equals( mAwayTeam ) )
+        {
+            mHasBallPossession = mAwayTeam;
+        }
+        else if( team.equals( mHomeTeam ) )
+        {
+            mHasBallPossession = mHomeTeam;
+        }
+    }
+
+    public void startNewEvent()
+    {
+        mEventGameClock = mCurrentGameClock;
+    }
+
+    public void swapBallPossession()
+    {
+        if( mHasBallPossession.equals( mHomeTeam ) )
+        {
+            mHasBallPossession = mAwayTeam;
+        }
+        else
+        {
+            mHasBallPossession = mHomeTeam;
+        }
+        resetShotClock24();
+    }
+
+    public void unpauseGame()
+    {
+        mGameOngoing = true;
+    }
+
+    public void updateTime()
+    {
+        if( isPeriodOngoing() && isPossessionOngoing() && isGameOngoing() )
+        {
+            mHasBallPossession.updatePossessionTime();
+            mAwayTeam.updatePlayingTime();
+            mHomeTeam.updatePlayingTime();
+            mCurrentGameClock--;
+            mCurrentShotClock--;
+        }
+        else
+        {
+            pauseGame();
+        }
+    }
+
+    private void addPeriodFoul( Team team )
+    {
+        if( team.equals( mAwayTeam ) )
+        {
+            mAwayPeriodFouls++;
+        }
+        else if( team.equals( mHomeTeam ) )
+        {
+            mHomePeriodFouls++;
+        }
+    }
+
+    private void checkTeamFoulEvent( GameEvent event )
+    {
+        if( event == null )
+        {
+            return;
+        }
+        if( event instanceof ShootingFoulEvent || event instanceof NonShootingFoulEvent )
+        {
+            addPeriodFoul( event.getTeam() );
+        }
+        else
+        {
+            checkTeamFoulEvent( event.getAppended() );
+        }
+    }
+
+    private void checkTurnoverEvent( GameEvent event )
+    {
+        if( event == null )
+        {
+            return;
+        }
+        if( event instanceof TurnoverEvent )
+        {
+            swapBallPossession();
+        }
+    }
+
+    private void initializeClocks()
+    {
+        pauseGame();
+        resetGameClock();
+        resetShotClock24();
+    }
+
+    private void initializeLogs()
+    {
+        mGameLog = new GameLog();
+        mPeriodLog = mGameLog.getCurrentPeriodLog();
+        mPeriod = mGameLog.getCurrentPeriod();
+    }
+
+    private void resetPeriodFouls()
+    {
+        mHomePeriodFouls = 0;
+        mAwayPeriodFouls = 0;
+    }
+
+    public enum GameStats
+    {
+
+        ID( "id" ), AWAY_TEAM( "awayTeam" ), HOME_TEAM( "homeTeam" ), GAME_LOG( "gameLog" ), PERIOD_LOG(
+                "periodLog" ), DATE( "date" );
+
+        private final String mConstant;
```

Comment:
You can instead make this a public variable, since this is already declared as final there will be no confusion or accident in which the value might change.